src/queries/completion.ts

```diff
@@ -0,0 +1,40 @@
+import type { Model, Prompt } from '../types';
+
+export type CompletionOptions = {
+  model: Model;
+  prompt: Prompt;
+};
+
+export class Completion {
+  private model: Model;
+  private prompt: Prompt;
+
+  constructor(options: CompletionOptions) {
+    this.model = options.model;
+    this.prompt = options.prompt;
+  }
+
+  async run(query: string) {
+    const prompt = await this.prompt.render({ query });
+
```

Comment:
As a question, if we wanted to add logging: would you add it here? Or in `model.run(prompt)` ?